### PR TITLE
Add new extension: dlmoji

### PR DESCRIPTION
**Pull request by Hydrapse**
## Description

An intelligent Emoji Dictionary helps to get related Emojis, Phrases, and AI-powered Translations in Raycast.
### Features
1. Supports Chinese & English Emoji Search
2. Supports Semantic, Verbatim, or Blended Emoji Translation
3. Gets related Emojis 🧐 or Emoji Phrases  🛌🏃‍♀️ with Explanations

## Screencast

dlmoji-1

dlmoji-3

Config-Panel

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by Hydrapse** [comment]
Hi @danielsequeira,
Sorry for bothering you. I'm wondering if something goes wrong, or something I can do to progress the review.
Looking forward to your reply!

**Comment by danielsequeira** [comment]
@Hydrapse one question, why the name DLmoji?

**Comment by danielsequeira** [comment]
On another note, I think we will need to improve the UX a bit on the extension. In cases where the emoji description is too long, it doesn't look good to render it in the action panel and there are cases where even the action panel doesn't have enough space for it:
dlmoji 2022-07-27 at 13 28 09

I would suggest for you to use our detail component: https://developers.raycast.com/api-reference/user-interface/detail where we could have a detail section for each emoji and allow the entire description to be rendered.

**Comment by Hydrapse** [comment]
> @Hydrapse one question, why the name DLmoji?

That's because I follow the nomenclature of [deepmoji](https://github.com/bfelbo/DeepMoji).
DLmoji means emoji search powered by Deep Learning or comparable intelligent APIs, which can handle phrase or sentence level queries, different from the existing [Emoji Search](https://www.raycast.com/FezVrasta/emoji) extension.

**Comment by danielsequeira** [comment]
> > @Hydrapse one question, why the name DLmoji?
> 
> That's because I follow the nomenclature of [deepmoji](https://github.com/bfelbo/DeepMoji). DLmoji means emoji search powered by Deep Learning or comparable intelligent APIs, which can handle phrase or sentence level queries, different from the existing [Emoji Search](https://www.raycast.com/FezVrasta/emoji) extension.

Got you! Thanks for the explanation. I think in that case it would probably make sense to mention deep learning in the package.json description and in the readme. I can imagine a lot of people like myself not really knowing what it means and that can affect the searches in the store for this extension

**Comment by Hydrapse** [comment]
ok, I got it. I'll update the description and try to improve the UX by following the advice 😉

**Comment by Hydrapse** [comment]
Sorry for being late. I've added the Detail Page for Emoji description and updated the Readme & package.json description. 
Please have a look. Thanks.
image

File: extensions/dlmoji/src/dlmoji.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        const hasChinese = /[\u4E00-\u9FA5]+/g.test(queryText)
+        const lang = hasChinese ? "zh" : "en"
+        const dataList: ITranslateReformatResult[] = []
+
+        // 接口速度较慢
+        async function getDeepmoji(): Promise<ITranslateReformatResultItem | undefined> {
+            let enText: string = queryText
+            if (hasChinese) {
+                // const res = await fetchDeepl(queryText)
+                // enText = res.data.data
```
Comment (danielsequeira) [inline extensions/dlmoji/src/dlmoji.tsx:57]: please remove commented code

File: extensions/dlmoji/src/api.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            return
+        })
+}
+
+export async function fetchEmojiAll(queryText: string, fromLanguage: string): Promise<any> {
+    const lang = fromLanguage === "zh" ? "zh-hans" : fromLanguage
+
+    // let data = new FormData();
+    // data.append('keywords', queryText)
+    // data.append('lang', lang)
```
Comment (danielsequeira) [inline extensions/dlmoji/src/api.tsx:125]: please remove commented code

File: extensions/dlmoji/src/api.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    return axios
+        .post(
+            "https://www.emojiall.com/elasticsearch/getData.php",
+            querystring.stringify({
+                keywords: queryText,
+                lang: lang,
+            }),
+            {
+                headers: {
+                    // 'Content-Type': 'multipart/form-data',
```
Comment (danielsequeira) [inline extensions/dlmoji/src/api.tsx:136]: please remove commented code

File: extensions/dlmoji/src/TranslateResult.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    const dataLength = data?.length - 1
+    let finalData: string[] = data
+    if (limitResultAmount > 0 && dataLength >= limitResultAmount) {
+        finalData = data.slice(0, limitResultAmount - 1)
+        finalData.push(data[dataLength])
+    }
+
+    return finalData.map((text, idx) => {
+        return {
+            // title: finalData.length - 1 === idx && idx > 0 ? "All" : truncate(text),
```
Comment (danielsequeira) [inline extensions/dlmoji/src/TranslateResult.tsx:21]: please remove commented code

**Comment by danielsequeira** [comment]
@Hydrapse thanks for the changes, I just added a few more comments, once they are addressed we should be good to merge this.

**Comment by Hydrapse** [comment]
Thanks @danielsequeira. I've committed all your changes.

**Comment by danielsequeira** [comment]
@Hydrapse thanks for the changes. One last thing I forgot, since it is not possible to read the entire description while in the List view and because you now have the View Description option, I think it makes sense to remove the description text from the list items:
dlmoji 2022-08-22 at 13 43 20

**Comment by Hydrapse** [comment]
The API service of "[deepmoji](https://github.com/bfelbo/DeepMoji) Emotion Analysis" is right here (**for Test Only**): 

`http://106.55.173.219:18878/ `

I think it might benefit the final testing 😊.

image

**Review by danielsequeira — APPROVED** [review]

**Comment by danielsequeira** [comment]
All looks good @Hydrapse, I just noticed though that now that you removed the description from the list items, your screenshots are not correct, because they still show the description and this can confuse users. Could you update them? If you do it in the next few hours, we can still merge and publish this today 🙂

**Comment by Hydrapse** [comment]
Besides updating all the screenshots in README and metadata, I also add a new feature.

I've found another Emoji Translation API, which is a perfect substitute for the "18dao Chinese Verbatim" API. Then I merge this new API into the Emoji Translation Section, which works for "Chinese To Emoji" Translation and "Emoji To Text" Translation:
image
image

**Review by danielsequeira — APPROVED** [review]

**Pull request merged**